pex/pex_bootstrapper.py

```diff
@@ -36,6 +42,38 @@ def parse_path(path):
     )
 
 
+@attr.s(frozen=True)
+class InterpreterTest(object):
+    entry_point = attr.ib()  # type: str
+    pex_info = attr.ib()  # type: PexInfo
+
+    @property
+    def interpreter_constraints(self):
+        # type: () -> List[str]
+        return self.pex_info.interpreter_constraints
+
+    def test_resolve(self, interpreter):
+        # type: (PythonInterpreter) -> bool
+        """Checks if `interpreter` can resolve all required distributions for the PEX under test."""
+        with TRACER.timed(
+            "Testing {python} can resolve PEX at {pex}".format(
+                python=interpreter.binary, pex=self.entry_point
+            )
+        ):
+            from pex.environment import PEXEnvironment
+
+            pex_environment = PEXEnvironment.mount(
+                self.entry_point,
+                pex_info=self.pex_info,
+                target=LocalInterpreter.create(interpreter),
+            )
+            try:
+                pex_environment.resolve()
```

Comment:
FYI: This resolve test right here was what indirectly caused the need for #1768 and killing pkg_resources use. That used a bunch of pkg_resources.Requirement.parse and pkg_resources.Distribution finding and loading code and led to the boot-time vendor meta_path importer clash with Pex said same.